plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Simplify/Opts.hs

```diff
@@ -20,6 +21,7 @@ data SimplifyOpts name a = SimplifyOpts
   , _soConservativeOpts        :: Bool
   , _soInlineHints             :: InlineHints name a
   , _soInlineConstants         :: Bool
+  , _soInlineThreshold         :: Int
```

Comment:
I'd use `Size` newtype (from the `PlutusCore.Size`) because this is what threshold is converted to in the end.